File: 4107-find-missing-elements/find-missing-elements.java
class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        int n=nums.length;
        int [] fre=new int[101];
        Arrays.fill(fre,0);
        int m=10000;
        int mi=-1;
        for(int i=0;i<n;i++)
        {
             m=Math.min(nums[i],m);
             mi=Math.max(nums[i],mi);
            fre[nums[i]]++;
        }
        List<Integer>lis=new ArrayList<>();
        for(int i=m;i<=mi;i++)
        {
            if(fre[i]==0)
            {
                lis.add(i);
            }
        }
        return lis;
        
    }
}

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        int n =nums.size();
        vector<int>ans;
        int m =-1,ma=INT_MAX;
        map<int,int>mp;
        for(int i =0;i<n;i++){
            ma=min(ma,nums[i]);
            m=max(m,nums[i]);
            mp[nums[i]]++;
        }
        for(int i =ma;i<=m;i++){
            if(mp.find(i)==mp.end()){
                ans.push_back(i);
            }
        }
        return ans;

    }
};